cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/PluginInstantiator.java

```diff
@@ -63,17 +65,26 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(PluginInstantiator.class);
 
+  private final LoadingCache<Location, ClassLoader> locationClassLoaders;
   private final LoadingCache<ArtifactDescriptor, ClassLoader> classLoaders;
   private final InstantiatorFactory instantiatorFactory;
   private final File tmpDir;
+  private final File pluginDir;
   private final ClassLoader parentClassLoader;
 
   public PluginInstantiator(CConfiguration cConf, ClassLoader parentClassLoader) {
     this.instantiatorFactory = new InstantiatorFactory(false);
-
     File tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
       cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+    File prefix = cConf.get(Constants.AppFabric.MRTASK_PLUGIN_DIR) != null ?
```

Comment:
also, this setting should always have some value so we don't need the null check. we should be sure to set it in cdap-default if we are not.